Gems/ROS2/Code/Source/Camera/CameraPublishers.cpp

```diff
@@ -6,19 +6,27 @@
  *
  */
 
-#include "CameraConstants.h"
 #include "CameraPublishers.h"
+#include "CameraConstants.h"
 #include "CameraSensor.h"
 #include <ROS2/Communication/TopicConfiguration.h>
-#include <ROS2/Utilities/ROS2Names.h>
 #include <ROS2/ROS2Bus.h>
+#include <ROS2/Sensor/SensorConfiguration.h>
+#include <ROS2/Utilities/ROS2Names.h>
 
 namespace ROS2
 {
     namespace Internal
     {
         using TopicConfigurations = AZStd::unordered_map<CameraSensorDescription::CameraChannelType, TopicConfiguration>;
 
+        TopicConfiguration GetTopicConfiguration(const SensorConfiguration& sensorConfiguration, const AZStd::string& key)
+        {
+            auto ipos = sensorConfiguration.m_publishersConfigurations.find(key);
+            AZ_Assert(ipos != sensorConfiguration.m_publishersConfigurations.end(), "Missing key in topic configuration!");
+            return ipos->second;
```

Comment:
How about returning an [AZStd::expected<TopicConfiguation, AZStd::string>](https://en.cppreference.com/w/cpp/utility/expected) which can account for the case when the topic configuration doesn't exist in the map?
```suggestion
            auto ipos = sensorConfiguration.m_publishersConfigurations.find(key);
            return ipos != sensorConfiguration.m_publishersConfigurations.end()
                ? AZStd::expected<TopicConfiguation, AZStd::string>(ipos->second)
               : AZStd::expected<TopicConfiguation, AZStd::string>(AZStd::unexpect, "Missing key in topic configuration!");
```

Reply:
@lumberyard-employee-dm this function is internal only and a missing entry is a logic error (there should be no way for the user to break that). This is why I opted for AZ_Assert.

Reply:
Ok, in that case can the return statement just return a default constructed TopicConfiguration then if the logic error ever occurred internally? If it did occur and the release configuration is running the assert would not trigger and the iterator would be invalid

```suggestion
            auto ipos = sensorConfiguration.m_publishersConfigurations.find(key);
            AZ_Assert(ipos != sensorConfiguration.m_publishersConfigurations.end(), "Missing key in topic configuration!");
            return ipos != sensorConfiguration.m_publishersConfigurations.end() ? ipos->second : TopicConfiguration{};
```